### ISSUE TITLE
Split the display text into regions and display with multiple phantoms

### ISSUE THREAD

**Issue by timfjord** [TheSecEng/LSP-copilot#53]

**Comment by jfcherng** [comment]
I think he means inline completion.

**Comment by timfjord** [comment]
Yeah, the idea of this issue is to parse the completion text before displaying the ghost text.  
Currently, we show the entire string from the payload, but, when, for example, you are in the middle of the line and there is a trailing text, most of the time the ghost text is confusing
<img width="511" alt="image" src="https://user-images.githubusercontent.com/471335/188440550-de62c1f4-c2df-4dde-acd8-c7b864cc781e.png">
Like on the picture above, the `(` symbol shouldn't be part of the completion

I will add more details to the issue later on.

Also, most likely this issue is valid only for single-line completion.

**Comment by TheSecEng** [comment]
I'll take a stab at this

**Comment by timfjord** [comment]
There is also an extreme case when you a completion should be split into multiple regions.  
I will try to send a screenshot when I reproduce it

**Comment by timfjord** [comment]
@TheSecEng I've added some scenarios to the card description.
